opengrok-tools/src/test/python/test_mirror.py

```diff
@@ -109,3 +114,59 @@ def test_valid_config():
     assert check_configuration({PROJECTS_PROPERTY:
                                 {"foo": {PROXY_PROPERTY: True}},
                                 PROXY_PROPERTY: "proxy"})
+
+
+@pytest.mark.skipif(not os.name.startswith("posix"), reason="requires posix")
+def test_incoming_retval(monkeypatch):
+    """
+    Test that the special CONTINUE_EXITVAL value bubbles all the way up to
+    the mirror.py return value.
+    """
+
+    class MockResponse:
+
+        # mock json() method always returns a specific testing dictionary
+        @staticmethod
+        def json():
+            return {"indexed": "true"}
+
+        @staticmethod
+        def raise_for_status():
+            pass
+
+    with tempfile.TemporaryDirectory() as source_root:
+        repo_name = "parent_repo"
+        repo_path = os.path.join(source_root, repo_name)
+        cloned_repo_name = "cloned_repo"
+        cloned_repo_path = os.path.join(source_root, cloned_repo_name)
+        project_name = "foo"    # does not matter for this test
+
+        os.mkdir(repo_path)
+
+        def mock_get_repos(*args, **kwargs):
+            from opengrok_tools.scm.repofactory import get_repository
+
+            return [get_repository(cloned_repo_path,
+                                   "git", project_name,
+                                   None, None, None, None)]
+
+        def mock_get(*args, **kwargs):
+            return MockResponse()
+
+        # Clone a Git repository so that it can pull.
+        import pygit2
```

Comment:
any special reason why the import is here?

Reply:
Recently I realized that imports can be anywhere so I am a bit on the path to use it as much as I can (a little bit like the joy of variable declaration of C99 compared to C89). Will use it more sparingly once I get enough :-)

Reply:
One thing is that we use it consistently I think in these tools at the top of the file.
Second thing is that it gives benefit when the function with the import is not executed (so the import does not trigger) but in this case it is executed each time